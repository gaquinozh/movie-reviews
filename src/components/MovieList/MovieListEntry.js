import React from 'react';

export default function MovieListEntry(props) {

  const movie = props.movie;

  return(
      <div className="movie-list__entry">{movie.title}</div>
  );
}
