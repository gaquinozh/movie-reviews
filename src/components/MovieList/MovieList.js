import React, { Component } from 'react';
import MovieListEntry from './MovieListEntry';

export default class MovieList extends Component {

  render() {
    const movies = this.props.movies.map(movie =>
      (
        <div className="cell" key={movie.id}>
          <MovieListEntry  movie={movie} />
        </div>
      )
    );

    return(
      <div>
        <header>
          <h1>Recent Movies</h1>
        </header>
        <main className="grid-x grid-margin-x grid-margin-y small-up-2 medium-up-4">
          {movies}
        </main>
      </div>
    );
  }
}
