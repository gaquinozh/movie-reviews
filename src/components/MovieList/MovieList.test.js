import React from "react";
import MovieList from "./MovieList";
import { shallow } from "enzyme";

describe("MovieList", () => {
  it("renders without crashing", () => {
    const wrapper = shallow(<MovieList movies={[]} />);
    expect(wrapper).toMatchSnapshot();
  });
});
