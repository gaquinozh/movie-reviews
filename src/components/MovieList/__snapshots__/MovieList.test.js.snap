// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MovieList renders without crashing 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <MovieList
    movies={Array []}
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <header>
          <h1>
                    Recent Movies
          </h1>
</header>,
        <main
          className="grid-x grid-margin-x grid-margin-y small-up-2 medium-up-4"
>
          
</main>,
      ],
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": <h1>
            Recent Movies
</h1>,
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Recent Movies",
          },
          "ref": null,
          "rendered": "Recent Movies",
          "type": "h1",
        },
        "type": "header",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [],
          "className": "grid-x grid-margin-x grid-margin-y small-up-2 medium-up-4",
        },
        "ref": null,
        "rendered": Array [],
        "type": "main",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <header>
            <h1>
                        Recent Movies
            </h1>
</header>,
          <main
            className="grid-x grid-margin-x grid-margin-y small-up-2 medium-up-4"
>
            
</main>,
        ],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": <h1>
              Recent Movies
</h1>,
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "Recent Movies",
            },
            "ref": null,
            "rendered": "Recent Movies",
            "type": "h1",
          },
          "type": "header",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [],
            "className": "grid-x grid-margin-x grid-margin-y small-up-2 medium-up-4",
          },
          "ref": null,
          "rendered": Array [],
          "type": "main",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
